ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -489,15 +543,17 @@ impl<'a> Assemble<'a> {
                     [RegConstraint::InputOutput(lhs), RegConstraint::Input(rhs)],
                 );
                 match size {
-                    1 => dynasm!(self.asm; and Rb(lhs_reg.code()), Rb(rhs_reg.code())),
-                    2 => dynasm!(self.asm; and Rw(lhs_reg.code()), Rw(rhs_reg.code())),
-                    4 => dynasm!(self.asm; and Rd(lhs_reg.code()), Rd(rhs_reg.code())),
-                    8 => dynasm!(self.asm; and Rq(lhs_reg.code()), Rq(rhs_reg.code())),
+                    1..=8 => {
```

Comment:
Why until `8`?

Reply:
Because once you go beyond 8 bytes, you can't sore the value in a register any more and you'd have to do something different.

(Would it have helped if the `size` were called `byte_size`?)

Reply:
The more I work on this system, the more I think variables and methods talking about a size should mention the units.

I'll admit I've confused bits for bytes in past. It's easily done.

Reply:
> Would it have helped if the size were called byte_size?

Yes it would!

Reply:
Fixed in 70b5ab8e53dd41d1c07a0482f49c640420c7020d and 3b79aa5ae485ec050fd658a6277101b9e03479ff.

I'll leave the latter as a separate commit, since the rest of this PR is about integer binops.